### DISCUSSION TITLE
Confused about actor/getSnapshot behavior in docs

### DISCUSSION THREAD

**Discussion by bstro** [statelyai/xstate#3475] [Q&A] [answered]
In the docs, under the section about actors and [syncing and reading state](https://xstate.js.org/docs/guides/actors.html#syncing-and-reading-state), the docs advise that in order to read an actor's state, one must spawn the actor with `{sync: true}` as an option and then call `getSnapshot` on the ref.

However, the docs then proceed to give _two_ examples of reading an actor's state, one that uses `getSnapshot` on the ref, and another that directly references `someRef.state`. Are both valid? Is there any difference between calling `someRef.state` and calling `someRef.getSnapshot().state`? 
<img width="774" alt="image" src="https://user-images.githubusercontent.com/627290/181296397-7f53f724-9ce8-4707-b3c8-3d39df56ff71.png">

**Comment by Andarist** [comment, marked as answer]
I think that the docs are not quite correct/precise here. `sync` is only relevant for `someRef.state` and it ensures that this property is kept up to date. Note though that we plan to remove `sync` in the next major version. 

`someRef.getSnapshot()` should work regardless of the `sync` option.

**Reply by bstro** [reply]
Thanks for this info! For what it's worth, `getSnapshot` doesn't seem to provide up-to-date data in my app without using `sync: true`. Happy to try to gather more information/context if you are interested.

**Reply by Andarist** [reply]
More information would be very much appreciated - I've rechecked the code and, from what I can tell, `getSnapshot` should always just return the up-to-date value

**Reply by bstro** [reply]
repro. hope this helps!

Make sure to refresh the page after enabling/disabling `{sync: true}` in `appMachine`

https://stackblitz.com/edit/vitejs-vite-chmsiu?file=src%2FApp.tsx,src%2Fmachines%2FappMachine.ts%3AL36

**Reply by bstro** [reply]
I'd like to add to the above comment that if you checked the repro within about 30 minutes of me posting, I made some changes and accidentally broke the reproduction. If you check again, the repro is fixed, stable, and ready to look at :)

**Reply by bstro** [reply]
Perhaps another clue: while `getSnapshot` does appear to be dependent on `sync: true` (at least in my stackblitz), I stumbled across `useActor` which gives me fresh state regardless of whether `sync` is true or not. 

https://stackblitz.com/edit/vitejs-vite-dphxmx?file=src%2FApp.tsx%3AL7
